microk8s-resources/actions/enable.storage.sh

```diff
@@ -7,9 +7,16 @@ source $SNAP/actions/common/utils.sh
 echo "Enabling default storage class"
 run_with_sudo mkdir -p ${SNAP_COMMON}/default-storage
 
-declare -A map
-map[\$SNAP_COMMON]="$SNAP_COMMON"
-use_manifest storage apply "$(declare -p map)"
+"$SNAP/microk8s-enable.wrapper" dns
+"$SNAP/microk8s-enable.wrapper" helm3
+
+"$SNAP/microk8s-helm3.wrapper" repo add rimusz https://charts.rimusz.net
+"$SNAP/microk8s-helm3.wrapper" repo update
+"$SNAP/microk8s-helm3.wrapper" install hostpath-provisioner rimusz/hostpath-provisioner \
```

Comment:
If we decide this is he way to go we should move/fork the storage repo somewhere we own.

Reply:
1. We should probably fork the repository, and consider what to do with https://github.com/juju-solutions/hostpath-provisioner

2. I don't feel comfortable with the idea of requiring helm in order to enable the hostpath provisioner. This stems from the fact that we lazily pull in the helm3 binary at runtime, which complicates things in environments where external connectivity may be restricted. I would prefer rendering the templates with `helm3 template $NAME $CHARM`, and then maintain the templates instead. The storage provider is something that will barely ever change, no need to introduce an upstream dependency here.

3. The default helm installation uses `/mnt/hostpath`, which is wrong.

4. We probably also need to preserve the provisioner and the storage class name. This provisioner uses `hostpath`, whereas downstream have started depending on `microk8s.io/hostpath` (e.g. Juju). This point probably ties in with 1, looking at the code of our current hostpath-provisioner and the one from the helm chart, it is probably in our best interest to simply update the dependencies of https://github.com/juju-solutions/hostpath-provisioner